core/modules/replica/tools/qserv-replica-job-move.cc

```diff
@@ -0,0 +1,215 @@
+    std::cout
+        << collectionName << ":\n"
+        << "----------+----------+-----+-----------------------------------------\n"
+        << "    chunk | database | rep | workers\n";
+
+    unsigned int prevChunk = (unsigned int) -1;
+
+    for (auto const& chunkEntry: collection) {
+
+        unsigned int const& chunk = chunkEntry.first;
+        for (auto const& databaseEntry: chunkEntry.second) {
+
+            std::string const& database    = databaseEntry.first;
+            size_t      const  numReplicas = databaseEntry.second.size();
+
+            if (chunk != prevChunk)
+                std::cout
+                    << "----------+----------+-----+-----------------------------------------\n";
+
+            prevChunk = chunk;
+
+            std::cout
+                << " "   << std::setw(8) << chunk
+                << " | " << std::setw(8) << database
+                << " | " << std::setw(3) << numReplicas
+                << " | ";
+
+            for (auto const& replicaEntry: databaseEntry.second) {
+
+                std::string          const& worker = replicaEntry.first;
+                replica::ReplicaInfo const& info   = replicaEntry.second;
+
+                std::cout << worker << (info.status() != replica::ReplicaInfo::Status::COMPLETE ? "(!)" : "") << " ";
+            }
+            std::cout << "\n";
+        }
+    }
+    std::cout
+        << "----------+----------+-----+-----------------------------------------\n"
+        << std::endl;
+}
+
+/// Run the test
+bool test() {
+
+    try {
+
+        ///////////////////////////////////////////////////////////////////////
+        // Start the controller in its own thread before injecting any requests
+        // Note that omFinish callbak which are activated upon a completion
+        // of the requsts will be run in that Controller's thread.
+
+        replica::ServiceProvider::Ptr const provider   = replica::ServiceProvider::create(configUrl);
+        replica::Controller::Ptr      const controller = replica::Controller::create(provider);
+
+        controller->run();
+
+        ////////////////////
+        // Start replication
+
+        std::atomic<bool> finished{false};
+        auto job = replica::MoveReplicaJob::create(
+            databaseFamily,
+            chunk,
+            sourceWorker,
+            destinationWorker,
+            purge,
+            controller,
+            std::string(),
+            [&finished] (replica::MoveReplicaJob::Ptr job) {
+                finished = true;
+            }
+        );
+        job->start();
+
+        util::BlockPost blockPost(1000,2000);
+        while (not finished) {
+            std::cout
+                << "qserv-replica-job-move:"
+                << "  Controller::numActiveRequests: " << controller->numActiveRequests()
+                << ", MoveReplicaJob::state: " << job->state2string()
+                << std::endl;
+            blockPost.wait();
```

Comment:
`condition_variable::wait_until`

Reply:
BlockPost::wait() waits for a random number of milliseconds in a range passed to the constructor of class BlockPost. This spin lock allows me to report a progress of the operation.